x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/utils/utils.ts

```diff
@@ -224,18 +224,23 @@ export const swapActionIds = async (
  */
 export const migrateLegacyActionsIds = async (
   rules: PromiseFromStreams[],
-  savedObjectsClient: SavedObjectsClientContract
+  savedObjectsClient: SavedObjectsClientContract,
+  actionsClient: ActionsClient
 ): Promise<PromiseFromStreams[]> => {
   const isImportRule = (r: unknown): r is RuleToImport => !(r instanceof Error);
 
   const toReturn = await pMap(
     rules,
     async (rule) => {
-      if (isImportRule(rule)) {
+      if (isImportRule(rule) && rule.actions != null && !isEmpty(rule.actions)) {
+        const [systemActions, extActions] = partition<RuleAction>(
+          (action) =>
+            action.action_type_id != null && actionsClient.isSystemAction(action.action_type_id)
+        )(rule.actions);
```

Comment:
Can you add a comment to explain what is going on here? I understand that you are separating system actions from other actions, but only because I have the context of reviewing this PR. Maybe add an explanation of what is being done and why.